core/modules/user/store/actions.ts

```diff
@@ -74,16 +74,13 @@ const actions: ActionTree<UserState, RootState> = {
   async refresh ({ commit }) {
     const usersCollection = StorageManager.get('user')
     const refreshToken = await usersCollection.getItem('current-refresh-token')
-    const resp = await UserService.invalidateToken(refreshToken)
+    const newToken = await UserService.refreshToken(refreshToken)
 
-    if (resp.code === 200) {
-      commit(types.USER_TOKEN_CHANGED, {
-        newToken: resp.result,
-        meta: resp.meta ? resp.meta : null
-      }) // TODO: handle the "Refresh-token" header
+    if (newToken) {
```

Comment:
how much better is THAT! <3

Reply:
@andrzejewsky @patzick So there's a bug here. If a user's password changes on another browser, device, w/e., the refresh token value will fail to login. In such a case, Magento returns:
```
{
  "code": 401,
  "result": "The account sign-in was incorrect or your account is disabled temporarily. Please wait and try again later."
}
```
As per line 94 of core/data-resolver/UserService.ts, the response.result is returned into the newToken variable and commits the value "The account sign-in was incorrect or your account is disabled temporarily. Please wait and try again later." into the user's current-token variable. 

The correct behavior would be to check that magento's returned code 200 on line 94 and return the result, otherwise null. Of course maybe for the user it would be nice to show an error like "Authorization expired. Please login again."